src/structs/pdb.rs

```diff
@@ -998,7 +1044,35 @@ impl FromIterator<Model> for PDB {
 #[cfg(test)]
 #[allow(clippy::unwrap_used)]
 mod tests {
+    use crate::open_pdb;
+
     use super::*;
+    use std::path::Path;
+
+    #[test]
+    fn remove_model() {
+        let pdb_dir = Path::new(env!("CARGO_MANIFEST_DIR")).join("example-pdbs");
+
+        for entry in std::fs::read_dir(pdb_dir).unwrap() {
+            let entry = entry.unwrap();
+            let path = entry.path();
+            if path.extension().unwrap() != "pdb" {
+                continue;
+            }
+            let (pdb, _) = open_pdb(path.to_str().unwrap(), crate::StrictnessLevel::Loose).unwrap();
+
+            let model_count = pdb.model_count();
+            let mut test_pdb = pdb.clone();
+            test_pdb.remove_model(0);
+            assert_eq!(test_pdb.model_count(), model_count - 1);
+
+            let mut test_pdb = pdb.clone();
+            test_pdb.remove_all_models_except_first(); // calls remove_models_except, so no need to test that explicitly
```

Comment:
I do agree that it calls `remove_models_except` but I still think having a small test on a single structure removing a couple and leaving a handful others would be nice just to make sure the function does not break down in future refactorings.